emcc.py

```diff
@@ -460,18 +460,13 @@ def is_valid_abspath(path_name):
       def check_bad_eq(arg):
         assert '=' not in arg, 'Invalid parameter (do not use "=" with "--" options)'
 
-      # Defaults to not showing absolute path warnings
-      absolute_warning_shown = True
+      # Scan and strip emscripten specific cmdline warning flags
+      # This needs to run before other cmdline flags have been parsed, so that warnings are properly printed during arg parse
+      newargs = shared.WarningManager.capture_warnings(newargs)
+      absolute_warning_shown = False
 
       for i in range(len(newargs)):
-        # Scan for path warning flag in advance from other cmdline flags, so that it works even if -I or -L directives are present before this.
-        if newargs[i] == '-Wwarn-absolute-paths':
-          newargs[i] = ''
-          absolute_warning_shown = False
-        elif newargs[i] == '-Wno-warn-absolute-paths':
-          newargs[i] = ''
-          absolute_warning_shown = True
-        elif newargs[i] in ['-l', '-L', '-I']:
+        if newargs[i] in ['-l', '-L', '-I']:
```

Comment:
Not sure, that would probably be a question for @kripken but I think he's got limited availability for a few days.